lib/logstash/outputs/elastic_app_search.rb

```diff
@@ -7,14 +7,24 @@ class LogStash::Outputs::ElasticAppSearch < LogStash::Outputs::Base
   config_name "elastic_app_search"
 
   config :engine, :validate => :string, :required => true
-  config :host, :validate => :string, :required => true
+  config :host, :validate => :string
+  config :url, :validate => :string
   config :api_key, :validate => :password, :required => true
   config :timestamp_destination, :validate => :string
   config :document_id, :validate => :string
+  config :path, :validate => :string, :default => "/api/as/v1/"
 
   public
   def register
-    @client = Client.new(@host, @api_key.value)
+    if @host.nil? && @url.nil?
+      raise ::LogStash::ConfigurationError.new("Please specify either \"url\" (for self-managed) or \"host\" (for SaaS).")
+    elsif @host && @url
+      raise ::LogStash::ConfigurationError.new("Both \"url\" or \"host\" can't be set simultaneously. Please specify either \"url\" (for self-managed) or \"host\" (for SaaS).")
+    elsif @host
+      @client = Client.new(@host, @api_key.value)
```

Comment:
this is a bit confusing to me; if both `host` and `path` are provided, my reading of this code is that we ignore the `path`?

Reply:
it now errors out if both `host` and `path`:

```
% bin/logstash -f cfg
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
Sending Logstash logs to /tmp/logstash-7.1.0/logs which is now configured via log4j2.properties
[2019-06-06T12:53:31,401][WARN ][logstash.config.source.multilocal] Ignoring the 'pipelines.yml' file because modules or command line options are specified
[2019-06-06T12:53:31,411][INFO ][logstash.runner          ] Starting Logstash {"logstash.version"=>"7.1.0"}
[2019-06-06T12:53:32,518][ERROR][logstash.javapipeline    ] Pipeline aborted due to error {:pipeline_id=>"main", :exception=>#<LogStash::ConfigurationError: Failed to connect to App Search. The setting "path" is not compatible with "host". Use "path" only with "url".>, :backtrace=>["/Users/joaoduarte/elastic/logstash-plugins/logstash-output-elastic_app_search/lib/logstash/outputs/elastic_app_search.rb:37:in `register'", "org/logstash/config/ir/compiler/OutputStrategyExt.java:106:in `register'", "org/logstash/config/ir/compiler/AbstractOutputDelegatorExt.java:48:in `register'", "/private/tmp/logstash-7.1.0/logstash-core/lib/logstash/java_pipeline.rb:191:in `block in register_plugins'", "org/jruby/RubyArray.java:1792:in `each'", "/private/tmp/logstash-7.1.0/logstash-core/lib/logstash/java_pipeline.rb:190:in `register_plugins'", "/private/tmp/logstash-7.1.0/logstash-core/lib/logstash/java_pipeline.rb:445:in `maybe_setup_out_plugins'", "/private/tmp/logstash-7.1.0/logstash-core/lib/logstash/java_pipeline.rb:203:in `start_workers'", "/private/tmp/logstash-7.1.0/logstash-core/lib/logstash/java_pipeline.rb:145:in `run'", "/private/tmp/logstash-7.1.0/logstash-core/lib/logstash/java_pipeline.rb:104:in `block in start'"], :thread=>"#<Thread:0x64c32fe run>"}
[2019-06-06T12:53:32,538][ERROR][logstash.agent           ] Failed to execute action {:id=>:main, :action_type=>LogStash::ConvergeResult::FailedAction, :message=>"Could not execute action: PipelineAction::Create<main>, action_result: false", :backtrace=>nil}
[2019-06-06T12:53:32,697][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2019-06-06T12:53:37,807][INFO ][logstash.runner          ] Logstash shut down.
```